src/main/java/io/confluent/connect/jdbc/source/MultiColumnTimestampHelper.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright 2017 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.connect.jdbc.source;
+
+import io.confluent.connect.jdbc.util.JdbcUtils;
+import io.confluent.connect.jdbc.util.StringUtils;
+import java.sql.Timestamp;
+import java.util.List;
+import org.apache.kafka.connect.data.Struct;
+
+/**
+ * MultiColumnTimestampHelper handles generation of queries when multiple timestamp
+ * columns are specified. Based on the input order, it selects the first not null
+ * timestamp value to detect new or modified rows.
+ */
+public class MultiColumnTimestampHelper implements TimestampHelper {
+
+  private List<String> timestampColumns;
+
+  public MultiColumnTimestampHelper(List<String> timestampColumns) {
+    this.timestampColumns = timestampColumns;
+  }
+
+  @Override
+  public void addWhereClause(StringBuilder builder,
+      String quoteString, String incrementingColumn) {
+    List<String> quotedList = JdbcUtils.quoteList(timestampColumns, quoteString);
+    String quotedTimestampColumn = StringUtils.join(quotedList, ",");
+    builder.append(" WHERE COALESCE(");
+    builder.append(quotedTimestampColumn);
+    builder.append(") < ? AND ((COALESCE(");
+    builder.append(quotedTimestampColumn);
+    builder.append(") = ? AND ");
+    builder.append(JdbcUtils.quoteString(incrementingColumn, quoteString));
+    builder.append(" > ?");
+    builder.append(") OR COALESCE(");
+    builder.append(quotedTimestampColumn);
+    builder.append(") > ?)");
+    builder.append(" ORDER BY COALESCE(");
+    builder.append(quotedTimestampColumn);
+    builder.append("),");
+    builder.append(JdbcUtils.quoteString(incrementingColumn, quoteString));
+    builder.append(" ASC");
```

Comment:
This is more FYI.. But if we use the String concatenation operator (`+`), then the java compiler will optimize it to use StringBuilder. (ref: https://docs.oracle.com/javase/specs/jls/se7/html/jls-15.html#jls-15.18.1).